emptyStrCmp/emptyStrCmp.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package emptyStrCmp
+
+import (
+	"go/ast"
+	"go/token"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name: "emptyStrCmp",
+	Doc:  "check for len(s) == 0 and len(s) != 0 where s is string",
+	Run:  run,
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	for _, file := range pass.Files {
+		ast.Inspect(file, func(node ast.Node) bool {
+			switch n := node.(type) {
+			case *ast.BinaryExpr:
+				callExpr, ok := n.X.(*ast.CallExpr)
+				if !ok {
+					return true
+				}
+				idt, ok := callExpr.Fun.(*ast.Ident)
+				if !ok {
+					return true
+				}
+				if idt.Name == "len" && (n.Op == token.EQL || n.Op == token.NEQ || n.Op == token.GTR) {
+					if len(callExpr.Args) < 1 {
+						return true
+					}
+					arg0 := callExpr.Args[0]
+					typ, ok := pass.TypesInfo.Types[arg0]
+					if ok && typ.Type.String() == "string" {
+						bLit, ok := n.Y.(*ast.BasicLit)
+						if !ok {
+							return true
+						}
+						if bLit.Value == "0" {
+							switch n.Op {
+							case token.EQL:
+								pass.Reportf(callExpr.Pos(), "calling len(s) == 0 where s is string, please use s == \"\" instead")
+							case token.NEQ:
+								pass.Reportf(callExpr.Pos(), "calling len(s) != 0 where s is string, please use s != \"\" instead")
+							case token.GTR:
+								pass.Reportf(callExpr.Pos(), "calling len(s) > 0 where s is string, please use s != \"\" instead")
+							}
+							return false
```

Comment:
Should we add a case for `len(s) <= 0`? It may seem strange but it's actually used.

Reply:
it is indeed strange! I added case for <= 0 and it did found some 2-3 instances of such usage!